test-framework/common/src/main/java/io/quarkus/test/common/TestResourceManager.java

```diff
@@ -105,11 +107,14 @@ public void stop() {
             }
         }
 
-        for (QuarkusTestResourceLifecycleManager i : ServiceLoader.load(QuarkusTestResourceLifecycleManager.class)) {
-            testResourceRunners.add(i);
+        Collections.sort(testResourceRunners, new QuarkusTestResourceLifecycleManagerComparator());
+
+        for (QuarkusTestResourceLifecycleManager quarkusTestResourceLifecycleManager : ServiceLoader
+                .load(QuarkusTestResourceLifecycleManager.class)) {
+            testResourceRunners.add(quarkusTestResourceLifecycleManager);
         }
 
```

Comment:
Shouldn't you sort the list when they are all added?

Reply:
Good catch. A fixup commit incoming.